src/validations/rules/sar.sch

```diff
@@ -176,6 +176,84 @@
 
     </sch:pattern>
 
+    <sch:pattern
+        id="results">
+        <sch:rule
+            context="oscal:observation">
+            <sch:assert
+                diagnostics="has-method-TEST-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.7"
+                id="has-method-TEST"
+                role="error"
+                test="
+                    if (oscal:type = 'finding')
+                    then
+                        oscal:method = 'TEST'
+                    else
+                        true()">An observation of type 'finding' must also have a method of 'TEST'.</sch:assert>
+
+            <sch:let
+                name="assessmentAssetsComponents"
+                value="/oscal:assessment-results/oscal:result/oscal:local-definitions/oscal:assessment-assets/oscal:assessment-platform/oscal:uses-component/@component-uuid" />
+
+            <sch:let
+                name="relatedObservationUUIDs"
+                value="/oscal:assessment-results/oscal:result/oscal:finding/oscal:related-observation/@observation-uuid" />
+
+            <sch:assert
+                diagnostics="has-related-observation-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.7"
+                id="has-related-observation"
+                role="error"
+                test="
+                    if (oscal:type = 'finding' and oscal:method = 'TEST' and oscal:subject[@type = 'component'])
+                    then
+                        if (@uuid[. = $relatedObservationUUIDs])
+                        then
+                            true()
+                        else
+                            false()
+                    else
+                        true()">An observation with a subject of component type must have a uuid value that matches a
+                related-observation/@observation-uuid value.</sch:assert>
+
+            <sch:assert
+                diagnostics="has-assessment-component-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.7"
```

Comment:
Documentation reference should be "_Guide to OSCAL-based FedRAMP Security Assessment **Results** (SAR)_".